assets/src/edit-story/app/media/utils/getResourceFromLocalFile.js

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import getTypeFromMime from './getTypeFromMime';
+import getFirstFrameOfVideo from './getFirstFrameOfVideo';
+
+/**
+ * Generates a virtual image to pick metadata resource
+ *
+ * @param {Object} image File object
+ * @return {Promise} Resource object
+ */
+
+const getImageResource = (image) => {
+  const reader = new window.FileReader();
```

Comment:
`getImageResource` and `getVideoResource` have a lot of repeated logic in them. Is there any change of moving to open function with a conditional statement for video?

Reply:
I see your point. I will try to reuse the logic between the methods used internally on both. Thanks for point this out.